src/mcp_server_uyuni/server.py

```diff
@@ -24,7 +24,12 @@
 from mcp import LoggingLevel, ServerSession, types
 from mcp_server_uyuni.logging_config import get_logger, Transport
 
-mcp = FastMCP("mcp-server-uyuni")
+from .auth import AuthProvider
+
+AUTH_SERVER = os.environ.get("UYUNI_AUTH_SERVER")
+
+auth_provider = AuthProvider(AUTH_SERVER) if AUTH_SERVER else None
+mcp = FastMCP("mcp-server-uyuni", auth=auth_provider)
```

Comment:
using an auth makes the transport protocol be http?

Reply:
Yes, in the final code we should validate these kind of things in the config values and write warnings in the logs when there's conflicting stuff like if there's an IDP specified and also stdio is selected at the same time, etc.

Another example is that it should be either Uyuni user/pass values or external idP.

I'm not planning to mark this PR ready to merge any time soon, so it's only skeleton code.

Reply:
As long as we make it clear with docs, log entries and comments on the example config, I think we can let it in case people wanna test/develop the server with simple user/pass credentials.